### ISSUE TITLE
Expose Helmet config

### ISSUE THREAD

**Issue by bchalk101** [backstage/backstage#12287]
Currently, in the app-config .yaml, the only available Helmet config is Content-Security-Policy options. As mentioned here https://github.com/backstage/backstage/blob/master/packages/backend-common/src/service/lib/ServiceBuilderImpl.ts#L224, it would be useful to have more access to the Helmet options.

In our specific situation, the header Strict-Transport-Security is causing us issues, as we have not yet deployed Backstage with tls. 

## Feature Suggestion

Allow customization of helmet options in the app-config.yaml.

## Possible Implementation

In the app-config.yaml under backend add a key `httpSecurity` or `httpHeaderSecurity`, the values under this key would each relate to an option in Helmet ie `contentSecurityPolicy` or `hsts`, as a starting point they can have `false` or `true` values, with the defaults being the same to what is currently implemented. 

Example
```
backend:
  httpHeaderSecurity:
    hsts: false
    originAgentCluster: false
```

Happy to contribute this, if this is something that sounds good.

**Comment by freben** [comment]
Yeah it makes sense to open up this for a bit more configuration. I'll assign you to this then.

I think as we progress with the so called "backend system" work, we should probably make this easier to drive through code rather than just config. As in, making it easy to hook into the setup phase and wholesale replace or remove helmet, if you prefer something else or different behaviors. But that's a bit farther down the line!

**Comment by freben** [comment]
Oh and are you sure that it's HSTS biting you? If it could be CSP `Upgrade-Insecure-Request`, which is reported commonly by others, you can disable that in config:

```
backend:
  csp:
    upgrade-insecure-requests: false
```

**Comment by bchalk101** [comment]
TBH not sure - we have since deployed behind tls so we are no longer having the issue. 

Happy to put this on the back burner, as no longer an issue for us.
Unless you believe this would benefit others?

**Comment by freben** [comment]
Alright, thanks for reporting back!

Honestly you are now on a good path anyway, and we'd like to hold off on the design of these extra exposed fields until there's a strong need for them. So all good from our end as well. I'll close this for now, but thank you for offering!

**Comment by Remyapraveen** [comment]
> Oh and are you sure that it's HSTS biting you? If it could be CSP `Upgrade-Insecure-Request`, which is reported commonly by others, you can disable that in config:
> 
> ```
> backend:
>   csp:
>     upgrade-insecure-requests: false
> ```

Hi, I am setting this element to false before deploying the app because previously my app wasn't displaying any contents from the HTML page once after deploying, I was giving me a blank page. Now after following the above method its working fine.  But just a doubt, could anyone please help shed some light if it is safe in the long run and/or What are the  other alternatives. ?

**Comment by eatmore01** [comment]
@Remyapraveen
Я ТВОИ НОГИ ЦЕЛОВАЛ
